api/nodes/permissions.py

```diff
@@ -22,6 +24,34 @@ def has_object_permission(self, request, view, obj):
             return obj.can_edit(auth)
 
 
+class AdminOrPublic(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, User)), 'obj must be a Node, got {}'.format(obj)
```

Comment:
The assertion allows type Node or User, but the error message only references Node.